systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java

```diff
@@ -0,0 +1,403 @@
+import io.strimzi.api.kafka.model.KafkaUser;
+import io.strimzi.api.kafka.model.KafkaUserScramSha512ClientAuthentication;
+import io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListenerBuilder;
+import io.strimzi.api.kafka.model.listener.arraylistener.KafkaListenerType;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.annotations.ParallelNamespaceTest;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.KafkaTopicResource;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBasicExampleAdminClient;
+import io.strimzi.systemtest.resources.crd.kafkaclients.AdminClientOperations;
+import io.strimzi.systemtest.resources.operator.SetupClusterOperator;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.JobUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@Tag(REGRESSION)
+public class ThrottlingQuotaST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(ThrottlingQuotaST.class);
+    static final String NAMESPACE = "throttling-quota-cluster-test";
+    final String kafkaUsername = "test-quota-user";
+    final String classTopicPrefix = "quota-topic-test";
+
+    /**
+     * Test checks for throttling quotas set for user
+     * on creation & deletion of topics and create partition operations.
+     */
+    @ParallelNamespaceTest
+    @Tag(INTERNAL_CLIENTS_USED)
+    void testThrottlingQuotasCreateTopic(ExtensionContext extensionContext) {
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(NAMESPACE, extensionContext);
+        int topicsCountOverQuota = 500;
```

Comment:
You seem to create a user with quota 2 ... but if I read the code right, you test it with 500 topics. That seems a bit disproportionate. Why do you need to create 500 topics and not just 3?

Reply:
Because it's `withControllerMutationRate(2.0)`, which is very weirdly described in KIP-599. And it has nothing to do with absolute number of topics/operation. So I just put some "random" big number, which should trigger the throttling quota limits. 
I would be very happy, if someone can explain to me what is the minimum number, when should it hit (as it is also based on requests per time etc).

Reply:
Well, I guess without knowing why 500 topics is the right limit for rate=2, we do not know if:
* This test will be stable
* It really tests what it says it tests

Or?

Reply:
Well, initially it also worked with rate=1.0, and it triggers quota exception everytime.
As you can see, I use "belowQuota" topics count, which is 100, when creating/deleting topics which are "outside" of test scenario, and that always passes.